### ISSUE TITLE
Add methods for disabling and destroying peripherals

### ISSUE THREAD

**Issue by hannobraun** [braun-robotics/rust-lpc82x-hal#42]
So far, I haven't added any methods to disable peripherals again after they have been initialized. While this shouldn't be necessary to get stuff to work, it is important to save power.

My thoughts on the matter:
- It should be possible to enable peripherals from both the `Unknown` state and the (yet to be added) `Disabled` state, so it probably makes sense to add another trait for the state types, `CanBeEnabled` or similar, to prevent code duplication.
- `Initialized` could be renamed to `Enabled`.
- It's worth thinking about how this relates to peripheral resets. I haven't really looked into those and don't know how they work in detail.
- Do peripherals keep their state when they are disabled but not reset? If so, that needs to be taken into account.

**Comment by hannobraun** [comment]
There should also be methods to destroy peripherals, making them completely unusable via the HAL and returning the raw register mappings from the [lpc82x](https://crates.io/crates/lpc82x) crate instead. This would be very useful for cases where LPC82x HAL doesn't provide all required features yet.

**Comment by hannobraun** [comment]
I've added `disable` methods in #79. I've decided not to add `destroy` methods, for the following reasons:
- Right now it's not needed, as peripheral APIs just borrow the peripheral. Although I don't think it's going to stay that way, so this item is only here for the sake of clarity, not as an argument against `destroy`.
- It's not always possible, as the registers might have been split between parts of an API. This means the original peripheral struct can't just be returned, because fields have been moved out of it.
- It's no longer necessary. Back when I opened this issue, there was a single `Peripherals` struct that managed all peripheral APIs. Now peripheral APIs have to be constructed manually, meaning it's easier to circumvent HAL APIs, if necessary.
- If all else fails, there's always an `unsafe` way to get a peripheral struct.
